src/common/crypt.ts

```diff
@@ -51,21 +66,20 @@ const runOnSecretFiles = async (crypt: CR, filesArgs: string[] = []): Promise<Pr
     files = await getAllSecretFiles()
   }
   preCrypt()
+  const chunkSize = 5
+  const filesChunked = chunkArray(files, chunkSize)
+
   const eventEmitterDefaultListeners = EventEmitter.defaultMaxListeners
-  EventEmitter.defaultMaxListeners = files.length + 5
+  if (chunkSize + 2 > EventEmitter.defaultMaxListeners) EventEmitter.defaultMaxListeners = chunkSize + 2
```

Comment:
Why not say chunkSize = 7 instead of adding 2?

Reply:
Adding 2 to make sure that there is room for other parallel execution to be run

Reply:
I could implement a parallel queue, as suggested in this article: https://techsparx.com/nodejs/async/queue-processing.html

Which should be able to queue, but afaik (and I've done a masters in CS Parallel Programming - so I should know (I hope)) the developer always needs to make a decision / calculate the resources available.

So here I've decided arbitrarily to chunk the work per-5 basis (and add 2 parallel threads for overhead)
If I use a queue, the same thing basically happens, maybe sightly smarted that full on chunks. But in that article you also read that `concurren(t/cy)` needs to be set (they set it to 5 as well)

I could find how many CPU cores are available and use that - 1 or -2 (need some cores idle to be able to process the queue, results and other system resources) But if decryptions is indeed memory bound (I found that decrypting 5 files takes 40MB in execution - so I can't imagine it being memory bound) then I need to figure out how much decryption takes in memory, divide available memory, calculate and make an estimation. But then again, that also wastes computational cycles, where I (as a developer) need to make an assumption how much memory a single decryption takes.

As you can read in this write up, in parallel programming there are a lot of nuances to think of if you want to optimize of most efficient. But sometime parallel programming can be made easier by making hardcoded arbitrary assumptions (like 5 decryptions in parallel) because it is "good enough"

Now I'm happy to optimize this some more, and doing some calcuations, but I think that 5 would be a nice limit.

Another thing you mentioned, disabling parallel execution is definitely an option, but rewriting the CLI from bash to TS, was that it is easier tot parallelize execution. One of the use cases for parallel execution was this exact function en/decryptions of secrets file. So disabling parallelization means we lose the performance gain here. Which would be a shame tbh, especially since this seems to work. If you find that during KinD there are not enough resources, then I suggest we look into optimizing the chunkSize instead of disabling parallelization completely. Since a chunkSize of 1 would be sequential anyway.